File: Models/ProjectSummary.cs
namespace MostaqlK.Models;

/// <summary>
/// Lightweight representation of a project as it appears in the listing feed, before
/// enrichment. Backs the project card in <c>Features/Projects/Views/ProjectCard.xaml</c>.
/// </summary>
public sealed class ProjectSummary
{
    public long ProjectId { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Url { get; set; } = string.Empty;

    public string ClientName { get; set; } = string.Empty;

    public int PublishTimeNumber { get; set; }

    public string PublishTimeText { get; set; } = string.Empty;

    public int ProposalCount { get; set; }

    public string ProposalCountText { get; set; } = string.Empty;

    public string? Budget { get; set; }

    public int? DeliveryDays { get; set; }

    public string SkillsText { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public string? ProjectStatus { get; set; }

    public bool IsUnread { get; set; } = true;

    public EnrichmentStatus EnrichmentStatus { get; set; } = EnrichmentStatus.Pending;

    public DateTimeOffset DiscoveredAt { get; set; }

    /// <summary>
    /// Timestamp at which this project's enrichment fully completed (set by
    /// <see cref="MostaqlK.Infrastructure.Http.Parsers.DetailParser"/> the moment detail parsing
    /// succeeds). Null while the project is still pending/being enriched. This — not
    /// <see cref="DiscoveredAt"/> — is what the feed sorts by, so a project only jumps to the
    /// top of the list once its enrichment has genuinely finished.
    /// </summary>
    public DateTimeOffset? EnrichedAt { get; set; }
}

File: .repertoire/python/src/mostaql/models/project_summary.py
"""Listing-feed project model (C# Models/ProjectSummary).

Field defaults are load-bearing (plan §4.1 trap 22): ``is_unread=True``,
``enrichment_status=PENDING``, empty-string-not-null for text fields.
"""

from dataclasses import dataclass, field
from datetime import datetime

from mostaql.models.enrichment_status import EnrichmentStatus

__all__ = ["ProjectSummary"]


@dataclass(frozen=True, slots=True)
class ProjectSummary:
    """Lightweight representation of a listing-feed project before enrichment.

    ``discovered_at`` is required and must be tz-aware UTC (listing parser stamps
    UtcNow per card); ``enriched_at`` stays null until enrichment completes.
    """

    project_id: int
    title: str
    url: str = ""
    client_name: str = ""
    publish_time_number: int = 0
    publish_time_text: str = ""
    proposal_count: int = 0
    proposal_count_text: str = ""
    description: str = ""
    budget: str | None = None
    delivery_days: int | None = None
    skills_text: str = ""
    project_status: str | None = None
    is_unread: bool = True
    enrichment_status: EnrichmentStatus = EnrichmentStatus.PENDING
    discovered_at: datetime = field(kw_only=True)
    enriched_at: datetime | None = field(default=None, kw_only=True)
